languages/ruby/exercises/concept/numbers/assembly_line.rb

```diff
@@ -0,0 +1,9 @@
+class AssemblyLine
+  def self.production_rate_per_hour(speed)
```

Comment:
This exercise only depends on the `basics` exercise and that exercise does not explain what the `self` notation is. This could possibly be introduced in this exercise, or mentioned in the after document of the  basics exercise, or possibly in a another prerequisite. @iHiD any thoughts?

Reply:
If I remember correctly the idea was that using `self` in these exercises (`strings` and `floating-point-numbers` also have them) was ok as long as they're in the scaffold already and the student doesn't need to know what exactly they mean at this point. I can add an explanation about `self` but I'm afraid the exercise will get a bit bloated as it already covers two concepts.

Reply:
I agree with both of you. I'd rather not have self, but I'm happy for it get merged for now and for us to tidy up across the whole track later down the road, if the other exercises already have it.

Reply:
I hate the hand wavy use of `self` but I don't think it is avoidable, as though it is key in understanding Ruby, it is also a fairly complicated thing.  Over the last few decades, my experience in teaching Ruby tells me that this should be covered (the `self` subject) thoroughly when we talk about iteration via Enumerable where we start authoring blocks for that purpose, and again when we start talking about composition, inheritance and mixins.  So at the beginner level but later, and at the early intermediate level, if I must categorize the skill level.